rsocket-core/src/main/java/io/rsocket/util/BackpressureUtils.java

```diff
@@ -0,0 +1,29 @@
+package io.rsocket.util;
+
+import io.rsocket.internal.LimitableRequestPublisher;
+import io.rsocket.internal.SynchronizedIntObjectHashMap;
+
+public class BackpressureUtils {
+
+  public static void shareRequest(
+      long requested,
+      SynchronizedIntObjectHashMap<LimitableRequestPublisher> limitableSubscriptions) {
+    try {
+      if (limitableSubscriptions.isEmpty()) {
+        return;
+      }
+      Object[] activeSubscriptions;
+      activeSubscriptions = limitableSubscriptions.getValuesCopy();
```

Comment:
copy the values each time you read isn't exceptionable and will lead to bad performance.